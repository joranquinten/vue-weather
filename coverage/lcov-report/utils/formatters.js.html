<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for utils/formatters.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">utils</a> formatters.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>9/9</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>8/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>8/8</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36</td><td class="line-coverage quiet"><span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">const weatherFormatter = (weather, type) =&gt; {
  if (!weather) return undefined
  if (!type || type === 'current') {
    return {
      summary: weather.currently.summary,
      icon: weather.currently.icon,
      latitude: weather.latitude,
      longitude: weather.longitude,
      temperature: weather.currently.temperature,
      time: weather.currently.time * 1000
    }
  }
&nbsp;
  if (type === 'daily') {
    return {
      summary: weather.currently.summary,
      icon: weather.currently.icon,
      latitude: weather.latitude,
      longitude: weather.longitude,
      daily: weather.daily.data.reduce((report, day) =&gt; {
        report.push({
          summary: day.summary,
          time: day.time * 1000,
          temperatureHigh: day.temperatureHigh,
          temperatureLow: day.temperatureLow,
          icon: day.icon
        })
        return report
      }, []),
      time: weather.currently.time * 1000
    }
  }
}
&nbsp;
export { weatherFormatter }
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Apr 23 2019 08:46:21 GMT+0200 (Central European Summer Time)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
