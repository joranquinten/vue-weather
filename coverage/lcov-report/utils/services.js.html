<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for utils/services.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">utils</a> services.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">77.78% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>7/9</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">77.78% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>7/9</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import axios from 'axios'
&nbsp;
const getWeatherForecast = async location =&gt; {
  <span class="missing-if-branch" title="if path not taken" >I</span>if (!location) {
<span class="cstat-no" title="statement not covered" >    throw new Error('No location provided!')</span>
  }
&nbsp;
  const secret = process.env.VUE_APP_DARKSKY_API_KEY
  const endpoint = `https://api.darksky.net/forecast/${secret}/REPLACE_WITH_LAT,REPLACE_WITH_LON?lang=nl&amp;units=ca&amp;exclude=minutely,hourly,alerts,flags`
&nbsp;
  const localForecast = endpoint
    .replace('REPLACE_WITH_LAT', location.latitude)
    .replace('REPLACE_WITH_LON', location.longitude)
&nbsp;
  return axios
    .get(localForecast)
    .then(response =&gt; {
      return response.data
    })
    .catch(<span class="fstat-no" title="function not covered" >er</span>ror =&gt; {
<span class="cstat-no" title="statement not covered" >      throw new Error(error)</span>
    })
}
&nbsp;
export { getWeatherForecast }
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Apr 23 2019 08:46:21 GMT+0200 (Central European Summer Time)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
