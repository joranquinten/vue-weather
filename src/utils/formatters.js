const weatherFormatter = (weather, type) => {
  if (!weather) return undefined
  if (!type || type === 'current') {
    return {
      summary: weather.currently.summary,
      icon: weather.currently.icon,
      latitude: weather.latitude,
      longitude: weather.longitude,
      temperature: weather.currently.temperature,
      time: weather.currently.time * 1000
    }
  }

  if (type === 'daily') {
    return {
      summary: weather.currently.summary,
      icon: weather.currently.icon,
      latitude: weather.latitude,
      longitude: weather.longitude,
      daily: weather.daily.data.reduce((report, day) => {
        report.push({
          summary: day.summary,
          time: day.time * 1000,
          temperatureHigh: day.temperatureHigh,
          temperatureLow: day.temperatureLow,
          icon: day.icon
        })
        return report
      }, []),
      time: weather.currently.time * 1000
    }
  }
}

export { weatherFormatter }
