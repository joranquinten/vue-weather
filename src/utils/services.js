import axios from 'axios'

const getWeatherForecast = async location => {
  if (!location) {
    throw new Error('No location provided!')
  }

  const secret = process.env.VUE_APP_DARKSKY_API_KEY
  const endpoint = `https://api.darksky.net/forecast/${secret}/REPLACE_WITH_LAT,REPLACE_WITH_LON?lang=nl&units=ca&exclude=minutely,hourly,alerts,flags`

  const localForecast = endpoint
    .replace('REPLACE_WITH_LAT', location.latitude)
    .replace('REPLACE_WITH_LON', location.longitude)

  return axios
    .get(localForecast)
    .then(response => {
      return response.data
    })
    .catch(error => console.error(error))
}

export { getWeatherForecast }
