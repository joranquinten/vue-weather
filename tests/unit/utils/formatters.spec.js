import { weatherFormatter } from '../../../src/utils/formatters'

const mockWeather = {
  currently: {
    icon: 'partly-cloudy-day',
    summary: 'Licht bewolkt',
    temperature: 12.91,
    time: 1555403544
  },
  daily: {
    data: [
      {
        icon: 'icon-day-one',
        summary: 'Dag 1',
        temperatureHigh: 20.41,
        temperatureLow: 8.57,
        time: 1555365600000
      },
      {
        icon: 'icon-day-two',
        summary: 'Dag 2',
        temperatureHigh: 10.17,
        temperatureLow: -1.24,
        time: 1555452000000
      }
    ]
  },
  latitude: 51.613358399999996,
  longitude: 5.5310954
}

describe('weatherFormatter', () => {
  it('should return undefined when nothing provided', () => {
    expect(weatherFormatter()).not.toBeDefined()
  })

  it("should return the daily format when no type or 'current' is provided", () => {
    expect(weatherFormatter(mockWeather)).toMatchSnapshot()
    expect(weatherFormatter(mockWeather, 'current')).toMatchSnapshot()
  })

  it("should return the daily format when 'daily' is provided", () => {
    expect(weatherFormatter(mockWeather, 'daily')).toMatchSnapshot()
  })

  it('should return something when a nonexistent type is provided', () => {
    expect(weatherFormatter(mockWeather, 'DOES_NOT_EXIST')).not.toBeDefined()
  })
})
