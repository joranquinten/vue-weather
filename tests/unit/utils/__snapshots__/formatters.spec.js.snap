// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`weatherFormatter should return the daily format when 'daily' is provided 1`] = `
Object {
  "daily": Array [
    Object {
      "icon": "icon-day-one",
      "summary": "Dag 1",
      "temperatureHigh": 20.41,
      "temperatureLow": 8.57,
      "time": 1555365600000000,
    },
    Object {
      "icon": "icon-day-two",
      "summary": "Dag 2",
      "temperatureHigh": 10.17,
      "temperatureLow": -1.24,
      "time": 1555452000000000,
    },
  ],
  "icon": "partly-cloudy-day",
  "latitude": 51.613358399999996,
  "longitude": 5.5310954,
  "summary": "Licht bewolkt",
  "time": 1555403544000,
}
`;

exports[`weatherFormatter should return the daily format when no type or 'current' is provided 1`] = `
Object {
  "icon": "partly-cloudy-day",
  "latitude": 51.613358399999996,
  "longitude": 5.5310954,
  "summary": "Licht bewolkt",
  "temperature": 12.91,
  "time": 1555403544000,
}
`;

exports[`weatherFormatter should return the daily format when no type or 'current' is provided 2`] = `
Object {
  "icon": "partly-cloudy-day",
  "latitude": 51.613358399999996,
  "longitude": 5.5310954,
  "summary": "Licht bewolkt",
  "temperature": 12.91,
  "time": 1555403544000,
}
`;
